test/spec/modules/afpBidAdapter_spec.js

```diff
@@ -0,0 +1,302 @@
+  REQUEST_METHOD,
+  TEST_PAGE_URL,
+  IS_DEV, mediaTypeByPlaceType
+} from 'modules/afpBidAdapter.js'
+
+const placeId = '60e7039918047b3fae304850'
+const bidId = '2a67c5577ff6a5'
+const transactionId = '7e8515a2-2ed9-4733-b976-6c2596a03287'
+const imageUrl = 'https://rtbinsight.ru/content/images/size/w1000/2021/05/ximage-30.png.pagespeed.ic.IfuX4zAEPP.png'
+const placeContainer = '#container'
+const imageWidth = 600
+const imageHeight = 400
+const pageUrl = IS_DEV ? TEST_PAGE_URL : 'referer'
+const sizes = [[imageWidth, imageHeight]]
+const bidderRequest = {
+  refererInfo: { referer: pageUrl },
+}
+const mediaTypeBanner = { [BANNER]: {sizes: [[imageWidth, imageHeight]]} }
+const mediaTypeVideo = { [VIDEO]: {playerSize: [[imageWidth, imageHeight]]} }
+const commonParams = {
+  placeId,
+  placeContainer,
+}
+const commonParamsForInImage = Object.assign({}, commonParams, {
+  imageUrl,
+  imageWidth,
+  imageHeight,
+})
+const configByPlaceType = {
+  get [IN_IMAGE_BANNER_TYPE]() {
+    return cloneDeep({
+      mediaTypes: mediaTypeBanner,
+      params: Object.assign({}, commonParamsForInImage, {
+        placeType: IN_IMAGE_BANNER_TYPE
+      }),
+    })
+  },
+  get [IN_IMAGE_MAX_BANNER_TYPE]() {
+    return cloneDeep({
+      mediaTypes: mediaTypeBanner,
+      params: Object.assign({}, commonParamsForInImage, {
+        placeType: IN_IMAGE_MAX_BANNER_TYPE
+      }),
+    })
+  },
+  get [IN_CONTENT_BANNER_TYPE]() {
+    return cloneDeep({
+      mediaTypes: mediaTypeBanner,
+      params: Object.assign({}, commonParams, {
+        placeType: IN_CONTENT_BANNER_TYPE
+      }),
+    })
+  },
+  get [IN_CONTENT_VIDEO_TYPE]() {
+    return cloneDeep({
+      mediaTypes: mediaTypeVideo,
+      params: Object.assign({}, commonParams, {
+        placeType: IN_CONTENT_VIDEO_TYPE
+      }),
+    })
+  },
+  get [OUT_CONTENT_VIDEO_TYPE]() {
+    return cloneDeep({
+      mediaTypes: mediaTypeVideo,
+      params: Object.assign({}, commonParams, {
+        placeType: OUT_CONTENT_VIDEO_TYPE
+      }),
+    })
+  },
+  get [IN_CONTENT_STORY_TYPE]() {
+    return cloneDeep({
+      mediaTypes: mediaTypeBanner,
+      params: Object.assign({}, commonParams, {
+        placeType: IN_CONTENT_STORY_TYPE
+      }),
+    })
+  },
+  get [ACTION_SCROLLER_TYPE]() {
+    return cloneDeep({
+      mediaTypes: mediaTypeBanner,
+      params: Object.assign({}, commonParams, {
+        placeType: ACTION_SCROLLER_TYPE
+      }),
+    })
+  },
+  get [ACTION_SCROLLER_LIGHT_TYPE]() {
+    return cloneDeep({
+      mediaTypes: mediaTypeBanner,
+      params: Object.assign({}, commonParams, {
+        placeType: ACTION_SCROLLER_LIGHT_TYPE
+      }),
+    })
+  },
+}
+const getTransformedConfig = ({mediaTypes, params}) => {
+  return {
+    params: params,
+    sizes,
+    bidId,
+    bidder: BIDDER_CODE,
+    mediaTypes: mediaTypes,
+    transactionId,
+  }
+}
+const validBidRequests = Object.values(configByPlaceType).map(config => getTransformedConfig(config))
```

Comment:
I think this is the issue with IE? Object.values isn't supported in IE (https://caniuse.com/object-values). You will need to refactor this using object.keys or a different method

Reply:
These are tests, they will not be executed in IE. Moreover, gulp, when building production, must convert the code to the old standard.
There is not only an object.values, there are also "entries", as well as destructurization and dynamic keys for objects. But these are just tests and they will not get into the "prebid.js" assembly in the browser.